pkg/reconciler/revision/resources/deploy.go

```diff
@@ -228,6 +232,18 @@ func MakeDeployment(rev *v1.Revision,
 		return nil, fmt.Errorf("failed to create PodSpec: %w", err)
 	}
 
+	replicaCount := autoscalerConfig.InitialScale
+	ann, found := rev.ObjectMeta.Annotations[autoscaling.InitialScaleAnnotationKey]
+	if found {
+		initialScale, err := strconv.Atoi(ann)
+		if err != nil {
+			return nil, fmt.Errorf("failed to process initialScale annotation: %w", err)
+		}
+		if initialScale != 0 || autoscalerConfig.AllowZeroInitialScale {
+			replicaCount = int32(initialScale)
+		}
+	}
+
```

Comment:
this function is getting quite long. Would you mind extracting this into a helper.